src/headset/oculus_mobile.c

```diff
@@ -566,12 +560,45 @@ void bridgeLovrUpdate(BridgeLovrUpdateData *updateData) {
     luaL_unref (Lcoroutine, LUA_REGISTRYINDEX, coroutineStartFunctionRef);
     coroutineStartFunctionRef = LUA_NOREF; // No longer needed
   }
-  if (lua_resume(Lcoroutine, 0) != LUA_YIELD) {
+  bool haveRenderError = renderErrorRef != LUA_NOREF;
+  if (haveRenderError) {
+    lua_rawgeti(Lcoroutine, LUA_REGISTRYINDEX, renderErrorRef); // Pull t from registry
+    luaL_unref (Lcoroutine, LUA_REGISTRYINDEX, renderErrorRef);
+    renderErrorRef = LUA_NOREF;
+  }
+  if (lua_resume(Lcoroutine, haveRenderError ? 1 : 0) != LUA_YIELD) {
     __android_log_print(ANDROID_LOG_DEBUG, "LOVR", "\n LUA QUIT\n");
     assert(0);
   }
 }
 
+static void lovrOculusMobileDraw(int framebuffer, int width, int height, float *eyeViewMatrix, float *projectionMatrix) {
+  lovrGpuDirtyTexture();
+
+  CanvasFlags flags = {0};
+  Canvas *canvas = lovrCanvasCreateFromHandle(width, height, flags, framebuffer, 0, 0, 1, true);
+
+  Camera camera = { .canvas = canvas, .stereo = false };
+  memcpy(camera.viewMatrix[0], eyeViewMatrix, sizeof(camera.viewMatrix[0]));
+  mat4_translate(camera.viewMatrix[0], 0, -offset, 0);
+
+  memcpy(camera.projection[0], projectionMatrix, sizeof(camera.projection[0]));
+
+  lovrGraphicsSetCamera(&camera, true);
+
+  if (renderUserdata && renderErrorRef == LUA_NOREF) { // Don't render if a render error occurred
+    int fn = lovrHeadsetExtractRenderFn(renderUserdata);
+    lua_pushcfunction(L, luax_getstack);
+    lua_rawgeti(L, LUA_REGISTRYINDEX, fn);
+    if (lua_pcall(L, 0, 0, -2)) {
+      renderErrorRef = luaL_ref(L, LUA_REGISTRYINDEX); // save luax_getstack output
+    }
+  }
```

Comment:
I don't think the "core headset.renderTo" code should know about Lua details like this.  It's a little more painful, but this sort of thing should go in `api/headset.c`.  Two ideas for this:

- The error could be stored in a known key in the registry like `_lovrRenderError` or something that the lua_resume code could extract later, or
- The render callback could have its signature modified to return an error message/value that `bridgeLovrDraw` could maybe store for later.

For more background on this motivation: my goal is to only have Lua code/headers in `main.c`, `luax.c`, and the `api` folder.  So far it's been a decent way to separate things and it makes it easier to add other language bindings in the future.